modules/preview-react/form-field/stories/examples/GroupedInputs.tsx

```diff
@@ -0,0 +1,215 @@
+    label: 'Mushrooms',
+    checked: false,
+  },
+];
+
+const bannerStyles = createStyles({
+  position: 'absolute',
+  right: 0,
+});
+
+export const GroupedInputs = () => {
+  const [toppingsState, setToppingsState] = React.useState(toppings);
+  const [error, setError] = React.useState(undefined);
+  const [radioError, setRadioError] = React.useState(undefined);
+  const [showSuccess, setShowSuccess] = React.useState(false);
+
+  const [value, setValue] = React.useState<string>('');
+  const [formData, setFormData] = React.useState({
+    toppings: [],
+    crust: '',
+  });
+  const handleCheckboxCheck = id => {
+    if (error) {
+      setError(undefined);
+    }
+    setToppingsState(
+      toppingsState.map(item => (item.id === id ? {...item, checked: !item.checked} : item))
+    );
+  };
+
+  const handleRadioChange = (e: React.ChangeEvent) => {
+    if (radioError) {
+      setRadioError(undefined);
+    }
+    const target = e.currentTarget;
+    if (target instanceof HTMLInputElement) {
+      setValue(target.value);
+    }
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    if (value === '' && toppingsState.every(item => !item.checked)) {
+      setRadioError('error');
+      setError('error');
+    } else if (toppingsState.some(item => !item.checked) && value === '') {
+      setError(undefined);
+      setRadioError('error');
+    } else if (value !== '' && toppingsState.every(item => !item.checked)) {
+      setError('error');
+      setRadioError(undefined);
+    } else {
+      setError(undefined);
+      setRadioError(undefined);
+    }
+    if (
+      error === undefined &&
+      radioError === undefined &&
+      toppingsState.some(item => item.checked) &&
+      value !== ''
+    ) {
+      setShowSuccess(true);
+    }
+    setFormData({
+      toppings: toppingsState,
+      crust: value,
+    });
+  };
+
+  React.useEffect(() => {
+    const timeout = setTimeout(() => {
+      if (showSuccess) {
+        setShowSuccess(false);
+      }
+    }, 3000);
+
+    return () => clearTimeout(timeout);
+  }, [showSuccess]);
+
+  const handleReset = () => {
+    setFormData({toppings: [], crust: ''});
+    setError(undefined);
+    setValue('');
+    setRadioError('');
+    setShowSuccess(false);
+    setToppingsState(
+      toppingsState.map(item => {
+        return {...item, checked: false};
+      })
+    );
+  };
+
+  return (
+    <div>
+      <h3>Choose your pizza options</h3>
+      <AriaLiveRegion role="alert">
+        <div style={{display: 'flex', gap: '40px'}}>
+          {error || radioError ? (
+            <Banner isSticky hasError className={bannerStyles}>
+              <Banner.Label>
+                {error && radioError
+                  ? 'At least one topping and crust selection is required'
+                  : error
+                  ? 'You must choose at least one topping'
+                  : radioError
+                  ? 'You must choose a crust'
+                  : ''}
```

Comment:
Why not setting error message instead having error and radioError? It will be less validation